src/main/java/com/wanikani/api/v2/request/AssignmentsRequest.java

```diff
@@ -1,166 +1,47 @@
-
-        public Builder createdAt(Instant createdAt) {
-            return createdAt(createdAt.atZone(ZoneOffset.UTC));
-        }
-
-        public Builder createdAt(OffsetDateTime createdAt) {
-            return createdAt(createdAt.toZonedDateTime());
-        }
-
-        public Builder srsStages(List<Integer> srsStages) {
-            this.srsStages = srsStages;
-            return this;
-        }
-
-        public Builder srsStages(Integer... srsStages) {
-            return srsStages(Arrays.asList(srsStages));
-        }
-
-        public Builder unlocked(Boolean unlocked) {
-            this.unlocked = unlocked;
-            return this;
-        }
-
-        public Builder started(Boolean started) {
-            this.started = started;
-            return this;
-        }
-
-        public Builder passed(Boolean passed) {
-            this.passed = passed;
-            return this;
-        }
-
-        public Builder burned(Boolean burned) {
-            this.burned = burned;
-            return this;
-        }
-
-        public Builder resurrected(Boolean resurrected) {
-            this.resurrected = resurrected;
-            return this;
-        }
-
-        public Builder hidden(Boolean hidden) {
-            this.hidden = hidden;
-            return this;
-        }
-
-        @Override
-        public AssignmentsRequest build() {
-            StringBuilder queryString = super.queryString();
-            appendList(queryString, "subject_ids", subjectIds);
-            appendList(queryString, "levels", levels);
-            appendList(queryString, "subject_types", subjectTypes);
-            appendDate(queryString, "created_at", createdAt);
-            appendDate(queryString, "available_before", availableBefore);
-            appendDate(queryString, "available_after", availableAfter);
-            appendList(queryString, "srs_stages", srsStages);
-            append(queryString, "unlocked", unlocked);
-            append(queryString, "started", started);
-            append(queryString, "passed", passed);
-            append(queryString, "burned", burned);
-            append(queryString, "resurrected", resurrected);
-            append(queryString, "hidden", hidden);
-            return new AssignmentsRequest(queryString.toString());
-        }
+        return super.queryString()
+            .appendList("subject_ids", subjectIds)
+            .appendList("levels", levels)
+            .appendList("subject_types", subjectTypes)
+            .appendDate("created_at", createdAt)
+            .appendDate("available_before", availableBefore)
+            .appendDate("available_after", availableAfter)
+            .appendList("srs_stages", srsStages)
+            .append("unlocked", unlocked)
+            .append("started", started)
+            .append("passed", passed)
+            .append("burned", burned)
+            .append("resurrected", resurrected)
+            .append( "hidden", hidden)
+            .toString();
     }
```

Comment:
This is much cleaner. I like the fluent API and how the request classes actually store the parameters now.